File: ecommerce/src/main/java/com/bautista/model/Product.java
package com.bautista.model;

import lombok.Data;
import java.math.BigDecimal;
import java.util.Date;

@Data
public class Product {
    private Integer id;
    private String name;
    private String description;
    private String categoryName;
    private String imageFile;
    private String unitOfMeasure;
    private BigDecimal price;

    // Add new fields for expansion
    private String productType;  // "footwear", "bag", "clothing"
    private String subCategory;
    private String brand;
    private String color;
    private String gender;

    private Integer stockQuantity;
    private String sku;

    private Integer discountPercentage;
    private Boolean isFeatured;
    private Boolean isNewArrival;
    private Boolean isBestSeller;

    private Double rating;
    private Integer reviewCount;

    // Flexible attributes (will be parsed from JSON in ProductData)
    private Object attributes;  // Can be Map<String, Object> or specific attribute classes

    private String status;
    private Date createdDate;
    private Date lastUpdated;
}

File: product/src/app/model/product.ts
export class Product{
    id: number = 0;
    name: string = '';
    description: string = '';
    productType: string = 'footwear';  // 'footwear', 'bag', 'clothing'
    categoryName: string = '';
    subCategory: string = '';
    unitOfMeasure: string = '';
    price: number = 0;
    imageFile: string = '';
    additionalImages: string[] = [];
    
    brand: string = '';
    color: string = '';
    gender: string = '';
    
    stockQuantity: number = 0;
    sku: string = '';
    
    discountPercentage: number = 0;
    isFeatured: boolean = false;
    isNewArrival: boolean = false;
    isBestSeller: boolean = false;
    
    rating: number = 0;
    reviewCount: number = 0;
    
    attributes: any = {};  // Type-specific attributes
    specifications: any = {};
    
    status: string = 'Active';
    
    lastUpdated: Date = new Date();
    created: Date = new Date();
}